ams/server/src/main/java/com/netease/arctic/server/metrics/MetricRegistry.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *  *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.server.metrics;
+
+import com.google.common.collect.Maps;
+import com.netease.arctic.ams.api.metrics.*;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.function.Consumer;
+
+/** A registry of amoro metric. */
+public class MetricRegistry implements MetricSet {
+
+  private final List<MetricRegisterListener> listeners = new CopyOnWriteArrayList<>();
+
+  private final ConcurrentMap<MetricKey, Metric> registeredMetrics = Maps.newConcurrentMap();
+  private final Map<String, MetricDefine> definedMetric = Maps.newConcurrentMap();
```

Comment:
1. `definedMetric` should be in plural form, eg `definedMetrics`.

2. `definedMetric` currently only keeps adding elements to the Map, we should consider the conditions for removing elements, otherwise it may cause memory leak. 

One removal condition could be to use a counter to record the number of references to the MetricDefine, when registering a metric, the counter will be increased by one, when removing the metric, the counter will be decreased by one, and the MetricDefine can be removed when the counter's value is zero in `unregister` method. Thus the variable type could be `Map<String, Map<MetricDefine, Long>> definedMetric;`

Reply:
fixed.